pkg/api/v1alpha1/cloudstackmachineconfig_types.go

```diff
@@ -117,6 +117,51 @@ func (c *CloudStackMachineConfig) OSFamily() OSFamily {
 	return ""
 }
 
+func (c *CloudStackMachineConfigSpec) Equal(o *CloudStackMachineConfigSpec) bool {
+	if c == o {
+		return true
+	}
+	if c == nil || o == nil {
+		return false
+	}
+	if !c.Template.Equal(&o.Template) ||
+		!c.ComputeOffering.Equal(&o.ComputeOffering) {
+		return false
+	}
+	if c.Affinity != o.Affinity {
+		return false
+	}
+	if len(c.AffinityGroupIds) != len(o.AffinityGroupIds) ||
+		len(c.Users) != len(o.Users) ||
+		len(c.UserCustomDetails) != len(o.UserCustomDetails) {
+		return false
+	}
+	for i, groupId := range c.AffinityGroupIds {
+		if groupId != o.AffinityGroupIds[i] {
```

Comment:
This implies that they have to be the same order. Is that an okay assumption? We have implementations elsewhere where we use a map to compare the actual values. 

https://github.com/aws/eks-anywhere/blob/main/pkg/api/v1alpha1/cluster_types.go#L450

Reply:
neat, will refactor to use this, thank you